Tests/Unit/MSFT_xWaitForADDomain.Tests.ps1

```diff
@@ -0,0 +1,151 @@
+                $targetResource.DomainName | Should Be $testParams.DomainName
+            }
+
+            It "Returns an empty DomainName when domain is not found" {
+                Mock -CommandName Get-Domain -MockWith {}
+                $targetResource = Get-TargetResource @testParams
+                $targetResource.DomainName | Should Be $null
+            }
+        }
+        #endregion
+
+
+        #region Function Test-TargetResource
+        Describe "$($Global:DSCResourceName)\Test-TargetResource" {
+            It 'Returns a "System.Boolean" object type' {
+                Mock -CommandName Get-Domain -MockWith {return $fakeDomainObject}
+                $targetResource =  Test-TargetResource @testParams
+                $targetResource -is [System.Boolean] | Should Be $true
+            }
+
+            It 'Passes when domain found' {
+                Mock -CommandName Get-Domain -MockWith {return $fakeDomainObject}
+                Test-TargetResource @testParams | Should Be $true
+            }
+
+            It 'Fails when domain not found' {
+                Mock -CommandName Get-Domain -MockWith {}
+                Test-TargetResource @testParams | Should Be $false
+            }
+        }
+        #endregion
+
+
+        #region Function Set-TargetResource
+        Describe "$($Global:DSCResourceName)\Set-TargetResource" {
+            It "Doesn't throw exception and doesn't call Start-Sleep and Clear-DnsClientCache when domain found" {
+                Mock -CommandName Get-Domain -MockWith {return $fakeDomainObject}
+                Mock -CommandName Start-Sleep -MockWith {}
+                Mock -CommandName Clear-DnsClientCache -MockWith {}
+                {Set-TargetResource @testParams} | Should Not Throw
+                Assert-MockCalled -CommandName Get-Domain -Times 1 -Exactly -Scope It
+                Assert-MockCalled -CommandName Start-Sleep -Times 0 -Scope It
+                Assert-MockCalled -CommandName Clear-DnsClientCache -Times 0 -Scope It
+            }
+
+            It "Doesn't call Start-Sleep and Clear-DnsClientCache when domain found" {
+                Mock -CommandName Get-Domain -MockWith {return $fakeDomainObject}
+                Mock -CommandName Start-Sleep -MockWith {}
+                Mock -CommandName Clear-DnsClientCache -MockWith {}
+                {Set-TargetResource @testParams} | Should Not Throw
+                Assert-MockCalled -CommandName Start-Sleep -Times 0 -Scope It
+                Assert-MockCalled -CommandName Clear-DnsClientCache -Times 0 -Scope It
+            }
+
+            It "Throws exception when domain not found after $($testParams.RetryCount) retries" {
+                Mock -CommandName Get-Domain -MockWith {}
+                Mock -CommandName Start-Sleep -MockWith {}
+                Mock -CommandName Clear-DnsClientCache -MockWith {}
+                {Set-TargetResource @testParams} | Should Throw
```

Comment:
you should verify more than it just throws, but it's a good start.  `| should throw 'message' ` is a bit better